common/src/main/java/com/joyent/http/signature/KeyPairLoader.java

```diff
@@ -150,36 +179,69 @@ public static KeyPair getKeyPair(final byte[] pKeyBytes, final char[] password)
     /**
      * Read KeyPair from an input stream, optionally using password.
      *
-     * @param is private key content as a stream
+     * @param is       private key content as a stream
      * @param password password associated with key
      * @return public/private keypair object
      * @throws IOException If unable to read the private key from the string
      */
     public static KeyPair getKeyPair(final InputStream is,
-                              final char[] password) throws IOException {
+                                     final char[] password) throws IOException {
+        return getKeyPair(is, password, null);
+    }
+
+    /**
+     * Read KeyPair from an input stream, optionally using password.
+     *
+     * @param is       private key content as a stream
+     * @param password password associated with key
+     * @param provider security provider to use when loading the key
+     * @return public/private keypair object
+     * @throws IOException If unable to read the private key from the string
+     */
+    public static KeyPair getKeyPair(final InputStream is,
+                                     final char[] password,
+                                     final String provider) throws IOException {
```

Comment:
I'm not sure if this would be better as an Enum, but I think there should at least be an explicit check on the contents of `provider`?  I think as it is now any string other than `SunPKCS11-NSS` silenty means BouncyCastle.

Reply:
(And list valid values in javadoc.)

Reply:
An enum would simplify the unexpected-value-checking so I'll go with that.